stock_barcodes/wizard/stock_barcodes_read_picking.py

```diff
@@ -0,0 +1,293 @@
+        self._set_default_picking()
+
+    def action_done(self):
+        if self.check_done_conditions():
+            if self._process_stock_move_line():
+                return super().action_done()
+
+    def action_manual_entry(self):
+        result = super().action_manual_entry()
+        if result:
+            self.action_done()
+        return result
+
+    def _prepare_move_line_values(self, candidate_move):
+        return {
+            'picking_id': self.picking_id.id,
+            'move_id': candidate_move.id,
+            'qty_done': self.product_qty,
+            'product_uom_id': self.product_id.uom_po_id.id,
+            'product_id': self.product_id.id,
+            'location_id': self.picking_id.location_id.id,
+            'location_dest_id': self.location_id.id,
+            'lot_id': self.lot_id.id,
+            'lot_name': self.lot_id.name,
+        }
+
+    def _prepare_stock_moves_domain(self):
+        domain = [
+            ('product_id', '=', self.product_id.id),
+            ('state', '=', 'assigned'),
+            ('picking_id.picking_type_id.code', '=', self.picking_type_code),
+        ]
+        if self.picking_id:
+            domain.append(('picking_id', '=', self.picking_id.id))
+        return domain
+
+    def _set_candidate_pickings(self, candidate_pickings):
+        vals = [(5, 0, 0)]
+        vals.extend([(0, 0, {
+            'picking_id': p.id,
+        }) for p in candidate_pickings])
+        self.candidate_picking_ids = vals
+
+    def _search_candidate_pickings(self, moves_todo=False):
+        if not moves_todo:
+            moves_todo = self.env['stock.move'].search(
+                self._prepare_stock_moves_domain())
+        if not self.picking_id:
+            candidate_pickings = moves_todo.mapped('picking_id')
+            candidate_pickings_count = len(candidate_pickings)
+            if candidate_pickings_count > 1:
+                self.select_cadidate_picking(candidate_pickings)
+                return False
+            if candidate_pickings_count == 1:
+                self.picking_id = candidate_pickings
+                self._set_candidate_pickings(candidate_pickings)
+            _logger.info('No picking assigned')
+        return True
+
+    def _process_stock_move_line(self):
+        """
+        Search assigned stock moves from a picking operation type or a picking.
+        If there is more than one picking with demand from scanned product the
+        interface allow to select what picking to work.
+        If only there is one picking the scan data is assigned to it.
+        """
+        StockMove = self.env['stock.move']
+        StockMoveLine = self.env['stock.move.line']
+        moves_todo = StockMove.search(self._prepare_stock_moves_domain())
+        if not self._search_candidate_pickings(moves_todo):
+            return False
+        lines = moves_todo.mapped('move_line_ids').filtered(
+            lambda l: (l.picking_id == self.picking_id and
```

Comment:
Why not adding this to the initial domain prepared by `_prepare_stock_moves_domain`?

Reply:
Mybe I only need filter by lot